lib/process-services-cloud/src/lib/task/task-form/components/task-form-cloud.component.html

```diff
@@ -57,7 +57,7 @@ <h4>
                 (success)="onClaimTask()" (error)="onError($event)">
             {{'ADF_CLOUD_TASK_FORM.EMPTY_FORM.BUTTONS.CLAIM' | translate}}
         </button>
-        <button mat-button *ngIf="hasCandidateUsersOrGroups()" adf-cloud-unclaim-task [appName]="appName" [taskId]="taskId"
+        <button mat-button *ngIf="hasCandidateUsersOrGroups() && canUnclaimTask()" adf-cloud-unclaim-task [appName]="appName" [taskId]="taskId"
```

Comment:
what if we create directives?

Reply:
as we discussed, i used the same approach as it is for other buttons + i moved hasCandidateUsersOrGroups() check to the canUnclaimTask() to avoid multiple conditions in the view.